### DISCUSSION TITLE
How to refer to a file's title if it is not active?

### DISCUSSION THREAD

**Discussion by Gewerd-Strauss** [SilentVoid13/Templater#959] [Help]
Hello,

This is an issue I am having in response to a feature request for the Upcoming-plugin (cf. https://github.com/charliecm/obsidian-upcoming/pull/10).

# Short intro

The upcoming plugin generates daily notes in advance and opens them. I requested a means of generating the notes without opening them, because rendering them takes several seconds and I don't want to lock up my program for half a minute. My idea was to allow generation, but without opening the notes.  
The author charliecm gave a beta release in the linked-above PR, and I tested it. This step works flawlessly.

# The main issue: how to reference the date?

Now, the below contains a snippet as to how I currently get the date and weekday for the dailynote upon creation, so that I can insert the correct dates and subtemplates:

    ## Day Planner <% this.app.workspace.getActiveFile().basename %> - <% tp.date.now("dddd",0,this.app.workspace.getActiveFile().basename,"DD.MM.YYYY") %>
    ---
    ### Start of Day (<% this.app.workspace.getActiveFile().basename %>)
    <%  tp.file.include("{{Morning - 0" + tp.date.now("d dddd",0,this.app.workspace.getActiveFile().basename,"DD.MM.YYYY") + "}}") %>
    ---
    ### The Day (<% this.app.workspace.getActiveFile().basename %>)
    <%  tp.file.include("{{Day - 0" + tp.date.now("d dddd",0,this.app.workspace.getActiveFile().basename,"DD.MM.YYYY") + "}}") %>
    ---
    ### Evening (<% this.app.workspace.getActiveFile().basename %>)
    <%  tp.file.include("{{Evening - 0" + tp.date.now("d dddd",0,this.app.workspace.getActiveFile().basename,"DD.MM.YYYY") + "}}") %>
    ---
    
Now, this works flawlessly if the active file has the right name. However, as a consequence the filename of the active file is used, not the filename of the file in which the template is inserted/executed.

This means that if I generate a file for tomorrow (18.12.2022) via upcoming while working on todays file (17.12.2022), the template will refer to the 17th and insert the corresponding contents (if I don't render tomorrows file, as per the PR I've linked to at the start.)

Thus, the question I have is if there is a way to have the template refer to the file's name in the `tp.date.now(...)`-function calls, instead of leveraging `getActiveFile().basename`, so that templates in files created by `upcoming` refer to their respective dates.

Is this possible? 

Thank you.
Sincerely,
~Gw

**Comment by Zachatoo** [comment]
The owner recently commented on that PR, does the comment resolve this issue?

**Reply by Gewerd-Strauss** [reply]
Nope, it clarified an issue I had with an ambiguously-named setting and a command that was unclear to me - but it has no new information for this plugin's part of the problem.

**Comment by Zachatoo** [comment]
Instead of using `this.app.workspace.getActiveFile().basename`, can you use [tp.file.title](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/file-module.html#tpfiletitle)? That should refer to the title of the note that the template is run in.

**Reply by Gewerd-Strauss** [reply]
I don't recall the exact reason, only that there was one for why I am using this. However, given I don't even know myself, I think I'll take a closer look to replace with the simpler tp.file.title tomorrow. Upon closer inspection, it seems that the build of upcoming doesn't seem to execute tp at all. as in, a daily template containing even just a single `tp.file.title` won't execute tp. 

Hmm.
